Python/compile.c

```diff
@@ -3247,9 +3247,7 @@ compiler_try_except(struct compiler *c, stmt_ty s)
         excepthandler_ty handler = (excepthandler_ty)asdl_seq_GET(
             s->v.Try.handlers, i);
         SET_LOC(c, handler);
-        if (!handler->v.ExceptHandler.type && i < n-1) {
-            return compiler_error(c, "default 'except:' must be last");
-        }
+        assert(handler->v.ExceptHandler.type || i >= n-1);
```

Comment:
Do we also need to add this check to the AST validator? (since it might lead to crashes when compiling manually crafted AST objects).

Reply:
Yup, what makes me think that maybe this change is not worth it because I find this PR much more complex than what we currently have

Reply:
Before making a decision about this PR, could you please look at the parser change for except* on https://github.com/python/cpython/pull/29581 ?